documentation/git_training.md

```diff
@@ -0,0 +1,258 @@
+
+A helpful function git status prints out important information about your repo. Use it before and after most commands to understand how files have changed in your repo.
+
+```{bash}
+git status
+```
+
+We can see that the added text file is shown in red text. This tells us it is a change from how we cloned the repo and has not yet been prepared to be merged with our main repo.
+
+#### Adding and Committing
+
+To prepare our new file for the main repo we use git add which adds a file to Git's staging area. It tells Git, "Hey, look at this file." After adding the text file, if we check git status again the test_file.txt will now appear in green.
+
+```{bash}
+git add test_file.txt
+git status
+```
+
+Now that we have staged our file, we are ready to commit it. The command git commits commits changes made to added files to the repository. It says, "Hey, take a snapshot of the files I've added."
+
+```{bash}
+git commit -m "<message>"
+```
+
+Note: sometimes you may receive the message:
+
+type :quit <Enter> to quit VIM
+
+Please follow this [link](https://stackoverflow.com/questions/11828270/how-do-i-exit-vim) to see how you can resolve this issue.
+
+#### Pushing Changes
+
+We have now committed the changes and it's time to send them to our main repository. The command git push pushes your local commits to the remote repository on GitHub. It says, "hey, push all my snapshots to a shared folder so that other people can see them." The command below will send your changes to the GitHub page under a new branch but will not immediately change the main repo.
+
+```{bash}
+git push origin iss228
+```
+
+You will always follow up git push with "origin" and then the name of the branch you are working on ( in this case iss228).
+
+Before running the git push command, your code and commits could only be accessed on your local computer. After pushing the code, it will appear on the main repo as a pull request that needs to be reviewed.
+
+*Other helpful commands:*
+
+If you want to check what has been modified since the last commit you can use *git diff*. This will highlight which lines have been modified. Use the arrow keys to scroll, and press q if you need to exit the prompt.
+
+```{bash}
+git diff [filename] 
+```
+
+If you are like me, sometimes you will start working on the wrong branch or main branch. If you want to take the work that you have done and transition over to the correct branch you can use the code below.
+
+```{bash}
+git fetch
+git checkout --track origin/<issue name>
+```
+
+## Reviewing Code
+
+After changes are pushed from Git, GitHub via the mobility-from-poverty repository page, will show that a new pull request is ready.
+
+![](images/image-1895120824.png)
+
+![](images/image-267968374.png)
+
+After you submit work you should assign a reviewer (using the reviewers tab circled in red) to the pull request and create the pull request. This will notify that reviewer that your changes are ready to be looked at.
```

Comment:
Darn, is this where the training ends? I thought we are also having guidance for the reviewer. Maybe it is just me at this point that struggles with this. I have a bunch of notes that I've kept. When I follow the notes, it usually works, but sometimes I get errors that don't make sense because I just did it the same way. I guess I am not assigned to do this anymore anyway, so it is fine, but we should check how comfortable Anne is with review in GitHub because she's new and we have her doing review for all of Jay's metrics